trailmap/pathway_analysis.py

```diff
@@ -0,0 +1,319 @@
+    Requires edge attribute 'capacity'. Any edge without 'capacity' attribute
+    will be given infinite capacity and find_maximum_flow will not work as
+    intended.
+    '''    
+
+    warning_message = ("Warning: Maximum flow does not support " +
+        "MultiDiGraphs, the graph form that Trailmap uses. Graph will be " +
+        "converted to DiGraph, which can lose information if you have " +
+        "multiple edges running between two nodes. \nChecking if graph G " +
+        "has an identical DiGraph representation...\n")
+
+    if G.is_directed():
+        multiedges = has_multiedges(G)
+        if multiedges == True:
+            print(warning_message)
+            print("G appears to have multiple edges between at least one " +
+            "pair of nodes. Information WILL be lost in the switch to " +
+            "DiGraph. Proceeding with caution, consider the results at " +
+            "your own risk\n")
+            H = nx.DiGraph(G)
+        elif multiedges == False:
+            print(warning_message)
+            print("G does not appear to have multiple edges between any " +
+            "pair of nodes. No information should be lost in the switch to " +
+            "DiGraph. Proceeding with flow calculation.\n")
+            H = nx.DiGraph(G)
+        else:
+            H=G
+    else:
+        print("Provided graph is not a MultiDiGraph or DiGraph and other \
+        errors may occur. Abort.\n")
+        return None, None, G
+
+    max_flow_path = nx.maximum_flow(H, s, t)
+    max_flow = nx.maximum_flow_value(H, s, t)
+    return max_flow_path, max_flow, H
+
+
+def find_pathway_flow(G, pathway):
+    '''returns the maximum permissible flow for a given pathway. Any edge
+    without 'capacity' attribute will be given infinite capacity.
+    '''
+    multiedges = has_multiedges(G)
+    if multiedges != True:
+        # graph either has no multiedges, or is not in MultiDiGraph format
+        H = nx.DiGraph(G)
+        edges = set(pairwise(pathway))
+        #H.remove_nodes_from([n for n in G if n not in pathway])
+        H_sg = H.edge_subgraph(edges).copy()
+
+        sources = get_sources(H_sg)
+        sinks = get_sinks(H_sg)
+
+        # check for capacity in edges (if no capacity, NetworkX will raise an
+        # exception) and if every edge has infinite flow, return inf
+        infinite_flow = True
+        for edge in edges:
+            if 'capacity' not in H_sg.get_edge_data(*edge):
+                return math.inf
+            if math.isinf(H_sg.get_edge_data(*edge)['capacity']) is False:
+                infinite_flow = False
+            
+        if infinite_flow is True:
+            return math.inf
+        else:
+            path_flow = nx.maximum_flow_value(H_sg, sources[0], sinks[0])
+            return path_flow
+
+    else:
+        print("Provided pathway has multiple edges. Flow calculation across \
+               multi-edged pathways is currently unsupported. Returning None")
+        return None
```

Comment:
Note that I'm not supporting pathways that could be duplicated due to the existence of multiple edges between a pair of nodes

for example, the following graph
1 - 2 = 3 - 4
would result in two pathways,
- (1,2,3,4)
- (1,2,3,4)

They appear identical when listing the nodes of the pathway, but the (2,3) edge is different and may in fact represent different commodities. Right now, `acquisition_pathways` destroys duplicate paths because they're stored as a set. I raised this in issue #35, and I recommend that it be addressed in a future PR instead of this one because it will require changes to `acquisition_pathways` enumeration